ikarus/assembler/simpleAssemblers.inl

```diff
@@ -96,16 +126,18 @@ namespace Ikarus {
 
   template <typename Basis, typename FEContainer>
   Eigen::SparseMatrix<double> &SparseFlatAssembler<Basis, FEContainer>::getReducedMatrixImpl(
-      const FERequirementType &fErequirements) {
-    if (!isReducedOccupationPatternCreated) createReducedOccupationPattern();
-    if (!arelinearReducedDofsPerElementCreated) createlinearDofsPerElementReduced();
+      const FERequirementType &feRequirements) {
+    std::call_once(sparsePreProcessorReduced, [&](){
+      createReducedOccupationPattern();

```

Comment:
These functions should have the same interface as `createOccupationPattern` and the matrix should be passed